spirv_common.hpp

```diff
@@ -1415,12 +1415,12 @@ class Variant
 			group->pools[type]->free_opaque(holder);
 		holder = nullptr;
 
-		if (!allow_type_rewrite && type != TypeNone && type != new_type)
+		/*if (!allow_type_rewrite && type != TypeNone && type != new_type)
```

Comment:
Drive-by, as this is not my project(!). 

I'd prefer too see the code deleted entirely.
(In a pinch use #if 0 .. #endif)

Reply:
This is a sanity check to make sure an ID is not tagged with different types over the course of compiling. If this check triggers, it's most likely a bug.

Reply:
I remember this check being an issue during the rebase. It seems to be fine now.

Reply:
I re-enabled this check with the first update.